src/libraries/Common/tests/System/Collections/DebugView.Tests.cs

```diff
@@ -206,5 +217,32 @@ public static void TestDebuggerAttributes_Null(object obj)
             TargetInvocationException tie = Assert.Throws<TargetInvocationException>(() => Activator.CreateInstance(proxyType, (object)null));
             Assert.IsType<ArgumentNullException>(tie.InnerException);
         }
+
+        private class CustomKeyedCollection<TKey, TValue> : KeyedCollection<TKey, TValue> where TKey : notnull
+        {
+            public CustomKeyedCollection() : base()
+            {
+            }
+
+            public CustomKeyedCollection(IEqualityComparer<TKey> comparer) : base(comparer)
+            {
+            }
+
+            public CustomKeyedCollection(IEqualityComparer<TKey> comparer, int dictionaryCreationThreshold) : base(comparer, dictionaryCreationThreshold)
+            {
+            }
+
+            public Func<TValue, TKey> GetKeyForItemHandler { get; set; }
```

Comment:
Nit: I think some of that indirection could be removed if you just made the derived type non-generic and inlined the GetKeyBehavior in the implementation.